### PR TITLE
Add Broker takeover configuration

### PR DESCRIPTION
**Pull request by NickyMateev**
# Add Broker takeover configuration

In cases where takeover of platform brokers needs to be centrally disabled there needs to be a configuration property to do that.

### CODE REVIEW

File: pkg/sbproxy/reconcile/reconcile_brokers.go [hunk 1]
```diff
@@ -61,7 +61,9 @@ func (r *resyncJob) reconcileBrokers(ctx context.Context, existingBrokers, desir
 			platformBroker, shouldBeProxified := brokerKeyMap[getBrokerKey(desiredBroker)]
 
 			if shouldBeProxified {
-				r.updateBrokerRegistration(ctx, platformBroker.GUID, desiredBroker)
+				if r.options.TakeoverEnabled {
```
Comment (KirilKabakchiev) [inline pkg/sbproxy/reconcile/reconcile_brokers.go:65]: If you like the term takeover, could you unify the other alternative words used (such as proxifying, proxification, etc) to use takveOver as well

**Review by KirilKabakchiev — CHANGES_REQUESTED** [review]

**Comment by KirilKabakchiev** [comment]
Blacklist and takeover should be taken into account in notifications too !

File: pkg/sbproxy/reconcile/reconcile_brokers.go (hunk 1 above)
Comment (NickyMateev) [reply pkg/sbproxy/reconcile/reconcile_brokers.go:65]: Addressed.

File: pkg/sbproxy/notifications/handlers/broker_handler.go [hunk 2]
```diff
@@ -90,6 +94,11 @@ func (bnh *BrokerResourceNotificationsHandler) OnCreate(ctx context.Context, pay
 	brokerProxyPath := bnh.brokerProxyPath(brokerToCreate.Resource)
 	brokerProxyName := bnh.brokerProxyName(brokerToCreate.Resource)
 
+	if slice.StringsAnyEquals(bnh.BrokerBlacklist, brokerToCreate.Resource.Name) {
+		log.C(ctx).Infof("Broker with name %s is part of broker blacklist. Skipping notification...", brokerToCreate.Resource.Name)
```
Comment (KirilKabakchiev) [inline pkg/sbproxy/notifications/handlers/broker_handler.go]: what about notifications for visibilities for plans of brokers that are blacklisted ? They will come too.

File: pkg/sbproxy/notifications/handlers/broker_handler.go [hunk 3]
```diff
@@ -161,7 +175,7 @@ func (bnh *BrokerResourceNotificationsHandler) OnUpdate(ctx context.Context, pay
 	log.C(ctx).Infof("Successfully found platform broker with name %s and URL %s.", existingBroker.Name, existingBroker.BrokerURL)
 
 	if existingBroker.BrokerURL != brokerProxyPath {
-		log.C(ctx).Errorf("Platform broker with name %s has an URL %s and is not proxified by SM. No update will be attempted", existingBroker.Name, existingBroker.BrokerURL)
+		log.C(ctx).Errorf("Platform broker with name %s has an URL %s and is not taken over by SM. No update will be attempted", existingBroker.Name, existingBroker.BrokerURL)
```
Comment (KirilKabakchiev) [inline pkg/sbproxy/notifications/handlers/broker_handler.go:181]: well, if an update notification for broker that is blacklisted comes, what will happen ? Same for delete. Same for visibilities

**Review by KirilKabakchiev — CHANGES_REQUESTED** [review]

File: pkg/sbproxy/notifications/handlers/broker_handler.go (hunk 3 above)
Comment (NickyMateev) [reply pkg/sbproxy/notifications/handlers/broker_handler.go:181]: Update and Delete broker notifications will fall back to the case where we check the platform for the broker, but it won't be found, so nothing will happen:

https://github.com/Peripli/service-broker-proxy/blob/aac6c78095f2d61a4936cfbebf11964668b1b732/pkg/sbproxy/notifications/handlers/broker_handler.go#L172

https://github.com/Peripli/service-broker-proxy/blob/aac6c78095f2d61a4936cfbebf11964668b1b732/pkg/sbproxy/notifications/handlers/broker_handler.go#L236

File: pkg/sbproxy/notifications/handlers/broker_handler.go (hunk 2 above)
Comment (NickyMateev) [reply pkg/sbproxy/notifications/handlers/broker_handler.go]: Enable and disable access will fail as the broker for which access will try to be enabled/disabled won't exist (for both Create/Update vis notifications):

https://github.com/Peripli/service-broker-proxy/blob/aac6c78095f2d61a4936cfbebf11964668b1b732/pkg/sbproxy/notifications/handlers/visibilities_handler.go#L98

https://github.com/Peripli/service-broker-proxy/blob/aac6c78095f2d61a4936cfbebf11964668b1b732/pkg/sbproxy/notifications/handlers/visibilities_handler.go#L136

Comment (NickyMateev) [reply pkg/sbproxy/notifications/handlers/broker_handler.go]: Essentially, nothing should mess up, but we'll get some failure logs. I can check whether the broker for which access will be created/changed is in the blacklist or not?

Comment (KirilKabakchiev) [reply pkg/sbproxy/notifications/handlers/broker_handler.go]: Yes i know that we will get failure logs. The comment implied that in my opinion we should also handle /skip these notifications as we don;'t want false positive errors - maybe i did not form my comment well enough :)

Comment (NickyMateev) [reply pkg/sbproxy/notifications/handlers/broker_handler.go]: Alright, I'll handle it for both the broker and visibilities then

Comment (NickyMateev) [reply pkg/sbproxy/notifications/handlers/broker_handler.go]: Addressed.

File: pkg/sbproxy/notifications/handlers/broker_handler.go (hunk 3 above)
Comment (NickyMateev) [reply pkg/sbproxy/notifications/handlers/broker_handler.go:181]: Addressed.

**Comment by NickyMateev** [comment]
@KirilKabakchiev please re-review.

Travis started failing with:

```
pkg/sbproxy/notifications/handlers/broker_handler.go: 147:: cyclomatic complexity 11 of function (*BrokerResourceNotificationsHandler).OnUpdate() is high (> 10) (gocyclo)
The command "gometalinter --vendor ./..." exited with 1.
```

Fixed it by extracting one of the checks into a standalone function.

**Review by KirilKabakchiev — APPROVED** [review]

**Pull request merged**